app/service/file_svc.py

```diff
@@ -57,16 +63,23 @@ def __init__(self, plugins, exfil_dir):
 
     async def find_file_path(self, name, location=''):
         """
-        Find the location on disk of a file by name
+        Find the location on disk of a file by name. Will also return the path
+        to an encoded version of the file (encoded files share the same name
+        but end with the '.xored' extension).
         :param name:
         :param location:
         :return: a tuple: the plugin the file is found in & the relative file path
         """
+        encoded_name = '%s.xored' % (name,)
         for plugin in self.plugins:
             for root, dirs, files in os.walk('plugins/%s/%s' % (plugin, location)):
                 if name in files:
                     self.log.debug('Located %s' % name)
                     return plugin, os.path.join(root, name)
+                elif encoded_name in files:
```

Comment:
let's move this logic. in the 'download' function, we overwrite the 'name' when we call _compile. we should instead do:
```
name = await self._rename_file_requested(request.headers.get('file'), request.headers.get('platform'))
```
And inside _rename_file_requested, we can check _compile and _decode functions, both of which should return the "real name".